friendship/python/friendship.py

```diff
@@ -8,7 +8,7 @@
 from pslpython.rule import Rule
 
 MODEL_NAME = 'friendship'
-DATA_DIR = os.path.join('..', 'data', MODEL_NAME)
+DATA_DIR = os.path.join('..', 'data', MODEL_NAME, '0/eval')
```

Comment:
Break up `'0/eval'`, the whole point of using `os.path.join()` is to avoid using path separators (e.g. '/').

Reply:
Same applies for all the other paths.